source/client.c

```diff
@@ -2501,9 +2540,11 @@ static void s_unsubscribe_complete(
     struct unsubscribe_task_arg *task_arg = userdata;
 
     AWS_LOGF_DEBUG(AWS_LS_MQTT_CLIENT, "id=%p: Unsubscribe %" PRIu16 " complete", (void *)connection, packet_id);
-    if (task_arg->timeout_arg) {
-        task_arg->timeout_arg->cancelled = true;
+    if (task_arg->timeout_wrapper.timeout_task_arg) {
```

Comment:
document this madness
`/* NULL out pointers so that, when timeout task runs, it knows that the request is already complete */`